tools/mock-service-host/src/mid/oav/swaggerMocker.ts

```diff
@@ -65,6 +67,11 @@ export default class SwaggerMocker {
      */
     private patchResourceId(responses: any, liveRequest: LiveRequest) {
         const url = parseUrl(liveRequest.url)
+        const resourceTypeMatch = (url.pathname || '').match(/\/microsoft.[^/]+\/[^/]+\//i)
+        const resourceType = resourceTypeMatch
+            ? resourceTypeMatch[resourceTypeMatch?.length - 1].slice(0, -1)
```

Comment:
What's the purpose to do such patch for mock with examples?

Reply:
purpose is to pick resource "type" from visiting url. It was 'aaa...' before this patch, and this will be hit only when mocking a response instead of query-by-example-id mode